bagit.py

```diff
@@ -632,6 +645,23 @@ def _validate_bagittxt(self):
             if first_line.startswith(codecs.BOM_UTF8):
                 raise BagValidationError("bagit.txt must not contain a byte-order mark")
 
+    def _path_is_dangerous(self, path):
+        """
+        Return true if path looks dangerous, i.e. potentially operates
+        outside the bagging directory structure, e.g. ~/.bashrc, ../../../secrets.json,
+            \\?\c:\, D:\sys32\cmd.exe
+        """
+        # Disallow dots, slashes, and tildes as first character
+        disallow = r'.\/~'
+        if path[0] in disallow:
+            return True
+        # Disallow drive letters at beginning of path, e.g c:, D:
+        drive_pat = re.compile(r'^[A-Za-z]:')
+        if drive_pat.match(path):
```

Comment:
So focus on checking whether the resolved path is actually a child of the bag directory and avoid inspecting the path as a string altogether?